### PR TITLE
Ben's Spatial Small World Network Method

### PR DESCRIPTION
**Pull request by coryglover**

### CODE REVIEW

**Comment by coryglover** [comment]
@jkbren docstring added

File: netrw/rewire/spatial_small_worlds.py [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+            if verbose:
+                if does_remove:
+                    removed_edges[t] = [rand_edge[0],rand_edge[1]]
+                added_edges[t] = [randomList[0][0],randomList[0][1]]
+        if verbose:
+            return G, removed_edges, added_edges
+        else:
+            return G
+
+    def full_rewire(self,G,p,dim,alpha,copy_graph=False,is_periodic=True,does_remove=True,manhattan_dist=True,timesteps=-1,directed=False,verbose=False):
```
Comment (nwlandry) [inline netrw/rewire/spatial_small_worlds.py:99]: I think the default value for `copy_graph` in the `full_rewire` method is `True`.

File: netrw/rewire/spatial_small_worlds.py [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+            dimsize = 3
+        elif len(dim) == 2:
+            dimsize = 2
+        else:
+            raise ValueError("Lattice Dimension is not 2-3")
+        if dimsize == 3:
+            pos = {(x,y,z):(x+5*z/7,y+5*z/7) for x,y,z in G.nodes()}
+        elif dimsize == 2:
+            pos = {(x,y):(x,y) for x,y in G.nodes()}
+        nx.draw(G,pos)
```
Comment (nwlandry) [inline netrw/rewire/spatial_small_worlds.py:126]: very cool!

**Review by nwlandry — APPROVED** [review]
just a few small changes and good to push

**Pull request merged**